### PR TITLE
refactor: drop Transactional and ConnectionOrTransaction

### PR DESCRIPTION
**Pull request by ctron**
Directly use the `ConnectionTrait` from sea_orm.

### CODE REVIEW

File: modules/analysis/src/endpoints.rs [hunk 1]
```diff
@@ -34,12 +35,13 @@ pub fn configure(config: &mut utoipa_actix_web::service_config::ServiceConfig, d
...
 pub async fn analysis_status(
     service: web::Data<AnalysisService>,
+    db: web::Data<Database>,
     user: UserInformation,
     authorizer: web::Data<Authorizer>,
     _: Require<ReadSbom>,
 ) -> actix_web::Result<impl Responder> {
     authorizer.require(&user, Permission::ReadSbom)?;
-    Ok(HttpResponse::Ok().json(service.status(()).await?))
+    Ok(HttpResponse::Ok().json(service.status(&**db).await?))
```
Comment (jcrossley3) [inline modules/analysis/src/endpoints.rs:44]: I love concise code, but is this a tad too magic? I can't know at a glance what `&**db` is doing. Are we taking a reference of the result of calling first the `Deref` impl of `web::Data` and then that of `Database`? I'm not a rust noob, and I'm not sure. :smile: 

Is something like `service.status(db.get_ref().connection())` too wordy or less idiomatic?

Comment (ctron) [reply modules/analysis/src/endpoints.rs:44]: > Are we taking a reference of the result of calling first the `Deref` impl of `web::Data` and then that of `Database`?

Btw, close: The first `Deref` takes the `Arc` from `web::Data`, the second takes `T` (`Database`) from the `Arc`.

Comment (ctron) [reply modules/analysis/src/endpoints.rs]: Replaced this with `db.as_ref()`.

**Review by JimFuller-RedHat — APPROVED** [review]
took me a little while to get through, LGTM!

**Review by helio-frota — APPROVED** [review]
it is hard to see upload deadlock #999  using this refactor https://gist.github.com/helio-frota/931dc5cf93a850996b210dcf7f0003d8

**Review by jcrossley3 — APPROVED** [review]
